lib/internal/fs/utils.js

```diff
@@ -460,8 +460,11 @@ function nsFromTimeSpecBigInt(sec, nsec) {
 // converted to a floating point number, we manually round
 // the timestamp here before passing it to Date().
 // Refs: https://github.com/nodejs/node/pull/12607
+// if the value is negative, we round it another way
+// Refs: https://github.com/nodejs/node/pull/43714
 function dateFromMs(ms) {
-  return new Date(Number(ms) + 0.5);
+  const msNumeric = Number(ms);
+  return new Date(msNumeric + (msNumeric >= 0 ? 0.5 : -0.5));
```

Comment:
```suggestion
  return new Date(MathRound(msNumeric));
```
Rounding away from zero is the idea here, right?

Reply:
Yep, to counterweight rounding performed by `Date` constructor (I guess, calling it `trunc` will be more accurate than `floor` at this point). The original code was a more performant way to do it for non-negative numbers, but it shouldn't give speed benefit anymore.